UNITTESTS/README.md

```diff
@@ -0,0 +1,294 @@
+# Unit testing Mbed OS
+
+This document describes how to run and write unit tests for Mbed OS.
+
+## Prerequisites
+
+* GNU toolchains installed.
+    * GCC 6 or later
+    * MinGW-W64 GCC-6.4.0 or MinGW-W64 GCC-7.3.0 (Windows)
+* CMake 3.0+ installed.
+* Python 2.7.x or >3.5 and pip 10.0 (or newer) installed.
+* gcovr >=4.1
+
+### Installing dependencies on Debian/Ubuntu
+
+1. `sudo apt-get -y install build-essential cmake`
+2. Install python and pip:
+```
+sudo apt-get -y install python python-setuptools
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing dependencies on Mac OS
+
+1. Install [Homebrew](https://brew.sh/).
+2. Install gcc compilers and cmake with: `brew install gcc cmake`
+3. Install python and pip:
+```
+brew install python
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing dependencies on Windows
+
+1. Download and install [MinGW-W64](http://mingw-w64.org/).
+2. Download CMake binaries from https://cmake.org/download/ and run the installer.
+3. Download Python2.7 or Python3 from https://www.python.org/getit/ and run the installer.
+4. Add MinGW, CMake and Python into PATH.
+5. Install [gcovr](#installing-covr).
+6. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+### Installing covr
+
+Install gcovr code coverage tool globally with `pip install 'gcovr>=4.1'` or using virtualenv:
+
+#### virtualenv
+
+1. Install virtualenv if not installed with `pip install virtualenv`
+2. Install gcovr with:
+
+**[Debian/Linux/Mac OS]**
+```
+virtualenv pyenv
+. pyenv/bin/activate
+pip install 'gcovr>=4.1'
+```
+**[Windows]**
+```
+virtualenv pyenv
+pyenv\\Scripts\\activate
+pip install 'gcovr>=4.1'
```

Comment:
All looks good, well done! I give a quick try, everything works just fine, apart from this line.
Seems on my windows 10, pip not like the single quotes, but only works with double quotes. please confirm @lorjala

Reply:
Yes, single quotes do not work when using default command prompt, but they do with powershell.